subsys/modem/backends/Kconfig

```diff
@@ -22,11 +22,28 @@ config MODEM_BACKEND_UART_ASYNC
 	bool "Modem UART backend module async implementation"
 	default y if UART_ASYNC_API
 
+if MODEM_BACKEND_UART_ISR
+
+config MODEM_BACKEND_UART_ISR_RECEIVE_IDLE_TIMEOUT_MS
+	int "Modem ISR UART backend delay between first byte received and RECEIVE_READY pipe event sending"
+	default 20
+	help
+	  This defines the delay, in milliseconds, that the backend waits when receiving a byte before notifying that data is available.
+	  This is to avoid having the receiver call modem_pipe_receive() too often (e.g. for every byte).
+	  It temporarily disables the UART RX IRQ when swapping buffers which can cause byte loss at higher baud rates.
+	  Regardless of this value, when a receive buffer gets full the event to notify that data is available will be sent.
+	  A good value for this is is slightly less than (~90%) the time it takes to fill up one such buffer.
+	  It can be calculated as follows:
+	  (<UART receive_buf_size> / 2) / (<UART baud rate> / <UART bits per byte>) * <ms per sec>
+	  By default (for the modem_cellular driver): (512 / 2) / (115200 / 10) * 1000 = 22,222 => 20
```

Comment:
Very long lines here, I wonder if we could make the option title a bit shorter (to make it more user friendly when using `menuconfig` command).
Also the help text lines are quite log, perhaps they could be split to smaller pieces.